spark/src/main/scala/org/apache/spark/rdd/FilteredCartesianRDD.scala

```diff
@@ -28,13 +28,33 @@ import scala.reflect.ClassTag
 import org.apache.spark._
 import org.apache.spark.util.Utils
 
-
+/** Performs a cartesian join of two RDDs using filter and refine pattern.
+  *
+  * During RDD declaration n*m partitions will be generated, one for each possible cartesian mapping.
+  * During RDD execution summary functions will be applied in a map-side reduce to `rrd1` and `rdd2`.
+  * These results will be collected and filter using `metapred` for partitions with potential matches.
+  * Partition pairings with possible matches will be checked using `pred` in a refinement step.
+  * No shuffle will be performed for partition pairings not flagged by filter step.
```

Comment:
At the end of this sentence, I would just like to add 

> ... by the filter step, but of the metardds produced using the summary functions **will** be shuffled (as they must be).  The metardds contain one item per partition (a "bounding box" around the corresponding partition in the parent rdd), so it is assumed that this shuffle will be much less costly than shuffling the parent rdds.

or something like that.